### DISCUSSION TITLE
Issue with Delayed Playback of Playlists on AzuraCast

### DISCUSSION THREAD

**Discussion by BornKillaz** [AzuraCast/AzuraCast#7171] [Q&A]
Hello everyone,

I'm encountering a recurring issue on AzuraCast and would appreciate any insights or solutions from those who might have experienced something similar.

**Problem Description**:

When I have a playlist scheduled to play, it does not start at the scheduled time. Instead, AzuraCast insists on playing two queued tracks before starting the playback of the scheduled playlist. This disrupts the scheduling and causes frustration for both hosts and listeners.

**Current Configuration**:

Scheduling: I have set the playlist to play at a specific time.

Playlist Weight: The weight of the scheduled playlist is higher than the weight of the currently playing playlist.

Playlist Content: The playlist typically contains an mp3 file with a pre-recorded show, and the duration of the show can be one or two hours.

Despite those settings, the issue persists.

Has anyone else experienced this issue with AzuraCast?

What could be the cause of this problem?

Are there any specific settings or configurations that can ensure the playlist plays exactly at the scheduled time?

Any help or suggestions would be greatly appreciated. Thank you!

**Comment by gAlleb** [comment]
Do you set the interrupt option?
<img width="500" alt="Снимок экрана 2024-06-03 в 20 33 37" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/5da61f74-e951-4172-9ed8-4b8b2d736114">

> he playlist typically contains an mp3 file with a pre-recorded show, and the duration of the show can be one or two hours.

Do you use autocue? It may also happen that ffmpeg take too much time to analyze long track, thus another one kicks in to fill the gap, which is less likely.

> Playlist Weight: The weight of the scheduled playlist is higher than the weight of the currently playing playlist.

I shouldn't matter since scheduled playlists have more priority than general ones.

**Comment by silvaalex2021** [comment]
Just to know, in the radio settings, under the 'AutoDJ' tab, did you check the option 'use replaygain metadata'? If yes, then disable that and give it a try. I had the same issue and managed to solve it just by turning off this option!

I'm not sure if this analysis is taking too long as the friend mentioned in the comment above or if it's just some annoying bug with LS, and I say it could be LS because this error has been haunting me since the days of the old Centova Cast, and it always got resolved by disabling the use of replaygain metadata!

**Reply by gAlleb** [reply]
Indeed, replaygain can eat lots of resources if tracks are not pre-tagged with `replaygain_track_gain`. It's always better to pre-tag your tracks beforehand so replaygain could just read the tags instead of calculating dB values.

I believe `autocue`  works faster, but again - 2 hour track needs some time to calculate especially on a not very powerful machine.

**Reply by BornKillaz** [reply]
> Do you set the interrupt option? <img alt="Снимок экрана 2024-06-03 в 20 33 37" width="500" src="https://private-user-images.githubusercontent.com/144447266/336169944-5da61f74-e951-4172-9ed8-4b8b2d736114.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MzgwNzksIm5iZiI6MTcxNzQzNzc3OSwicGF0aCI6Ii8xNDQ0NDcyNjYvMzM2MTY5OTQ0LTVkYTYxZjc0LWU5NTEtNDE3Mi05ZWQ4LTRiOGIyZDczNjExNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QxODAyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNmRhNjViZjUxMzhhOGFiOWQyNDJjY2NmNmQ1Njk4MmRkYjNjYjNlNGE0OGU0YjUwYmJlNjFlZGRjMWU2NGViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hTRwpem0InQ_O-n2JF2u2F4tsb_FbqRF9eX-hojWonQ">
> 
Yes, I do have that setting checked. 

> Do you use autocue? It may also happen that ffmpeg take too much time to analyze long track, thus another one kicks in to fill the gap, which is less likely.
> 
No, I do not use autocue. Is that the new AzuraCast feature on the latest release? 

> > Playlist Weight: The weight of the scheduled playlist is higher than the weight of the currently playing playlist.
> 
> I shouldn't matter since scheduled playlists have more priority than general ones.
>
It's nice to know that. 



> Just to know, in the radio settings, under the 'AutoDJ' tab, did you check the option 'use replaygain metadata'? If yes, then disable that and give it a try. I had the same issue and managed to solve it just by turning off this option!
> 
> I'm not sure if this analysis is taking too long as the friend mentioned in the comment above or if it's just some annoying bug with LS, and I say it could be LS because this error has been haunting me since the days of the old Centova Cast, and it always got resolved by disabling the use of replaygain metadata!


Yes, I do use Reply Gain setting, and I would like to keep using it. What of I set AzuraCast to queue 4 songs in advance instead of 2, would that possible help giving some more time to process files, or would it make it even worst?

The VPS has 8GB RAM. 4Cores and only runs azuracast with one station and one website.

**Reply by silvaalex2021** [reply]
@gAlleb  Well, then this definitely consumes more resources than it should, because on all the servers I've used, from the simplest VPS to the dedicated one I'm using now, I've always had this problem, even with files as small as a 30-minute block. Sometimes I would receive the program in 4 blocks of 30 minutes, and even if everything was scheduled correctly, Liquidsoap always found a way to insert a random track before the start of the next block, and my programming schedule would go haywire!

**Reply by gAlleb** [reply]
> What of I set AzuraCast to queue 4 songs in advance instead of 2

It doesn't matter since it's a queue outside Liquidsoap and doesn't affect LS playback. Indeed, maybe you should disable replaygain. Or pre-tag your tracks with @moonbase59's Loudgain. 

I would suggest to try new `autocue` feature instead of build-in replaygain (in case your tracks are not pre-tagged), that has been introduced lately to Azuracast. 

If nothing helps you could use "Advanced playlists`. Edit your playlist to be advanced and edit LS configuration (3d box):

```radio = switch(id="schedule_switch", track_sensitive=false, [ ({ 12h0m-13h0m }, your_playlist), ({true}, radio) ])```
